sklearn/metrics/_scorer.py

```diff
@@ -969,24 +970,37 @@ def check_scoring(estimator=None, scoring=None, *, allow_none=False):
         The object to use to fit the data. If `None`, then this function may error
         depending on `allow_none`.
 
-    scoring : str, callable, list, tuple, or dict, default=None
+    scoring : str, callable, list, tuple, set, or dict, default=None
         Scorer to use. If `scoring` represents a single score, one can use:
 
         - a single string (see :ref:`scoring_parameter`);
         - a callable (see :ref:`scoring`) that returns a single value.
 
         If `scoring` represents multiple scores, one can use:
 
-        - a list or tuple of unique strings;
+        - a list, tuple or set of unique strings;
         - a callable returning a dictionary where the keys are the metric
           names and the values are the metric scorers;
-        - a dictionary with metric names as keys and callables a values.
+        - a dictionary with metric names as keys and callables a values. The callables
+          need to have the signature `callable(estimator, X, y)`.
 
         If None, the provided estimator object's `score` method is used.
 
     allow_none : bool, default=False
-        If no scoring is specified and the estimator has no score function, we
-        can either return None or raise an exception.
+        Whether to return None or raise an error if no `scoring` is specified and the
+        estimator has no `score` method.
+
+    raise_exc : bool, default=True
+        Whether to raise an exception if a subset of the scorers in multimetric scoring
+        fails or to return an error code.
```

Comment:
There is something weird here maybe the "to" should not be here.

Reply:
I see, do you mind inverting both part around the "or":
```
Whether to return an error code or to raise an ...
```

Having a really long sentence before the "or" make my parsing algorithm fail :).

Reply:
I can see what you mean.

However, I really think that with those kinds of speaking params (`raise_exc`) we should document what happens when `True` first, because this is the natural meaning (and the default). Otherwise we force people to switch signs while reading.

What about using parenthesis like so:

Whether to raise an exception (if a subset of the scorers in multimetric scoring fails) or to return an error code.

Reply:
The parenthesis make the trick for me.